### PR TITLE
fix qwen2vl vision eager-attention

### PR DESCRIPTION
**Pull request by simonJJJ**
# What does this PR do?

Fixes # ([issue](https://github.com/QwenLM/Qwen2-VL/issues/38))

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by simonJJJ** [comment]
@ArthurZucker @zucchini-nlp we made a quick fix of eager attention impl in qwen2vl.

File: src/transformers/models/qwen2_vl/modeling_qwen2_vl.py [hunk 1]
```diff
@@ -286,9 +287,9 @@ def forward(
         q = apply_rotary_pos_emb_vision(q.unsqueeze(0), rotary_pos_emb).squeeze(0)
         k = apply_rotary_pos_emb_vision(k.unsqueeze(0), rotary_pos_emb).squeeze(0)
 
-        attention_mask = torch.zeros([1, seq_length, seq_length], device=q.device, dtype=torch.bool)
+        attention_mask = torch.full([1, seq_length, seq_length], float("-inf"), device=q.device, dtype=torch.float)
```
Comment (ArthurZucker) [inline src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: the dtype should probably be the same as q.dtype. so 
```suggestion
        attention_mask = torch.full([1, seq_length, seq_length], torch.finfo(q.dtype).min, device=q.device, dtype=q.dtype)
```

**Review by ArthurZucker — COMMENTED** [review]
thanks 🤗

File: src/transformers/models/qwen2_vl/modeling_qwen2_vl.py (hunk 1 above)
Comment (12sunyan) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: I have tried both method, still the same error is there any other place need to be modified?

Comment (zhangfan-algo) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: ＋1

Comment (satorugojos) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: +1

Comment (GeLee-Q) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: > I have tried both method, still the same error is there any other place need to be modified?

Can you describe what kind of error it is?

Comment (GanLab123) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py:290]: +1

**Comment by simonJJJ** [comment]
Hi @ArthurZucker, commit as you suggested, sorry for the late reply bc my vacation week.

**Comment by ShuaiBai623** [comment]
Hi, @zucchini-nlp @ArthurZucker , If you have any further questions about this fix, feel free to let us know! 👀

**Review by zucchini-nlp — APPROVED** [review]
LGTM, thanks!

File: src/transformers/models/qwen2_vl/modeling_qwen2_vl.py (hunk 1 above)
Comment (ArthurZucker) [reply src/transformers/models/qwen2_vl/modeling_qwen2_vl.py]: yep you need both dtype and `torch.finfo(q.dtype).min`

**Review by ArthurZucker — APPROVED** [review]
LGTM!

**Comment by ArthurZucker** [comment]
Sorry everyone for merging a bit later 🤗

**Pull request merged**